doc/tutorial/search/index.md

```diff
@@ -0,0 +1,258 @@
+~~~console
+Index stored successfully.
+Index loaded successfully.
+~~~
+\endsolution
+
+# Search
+
+Using an index, we can now conduct searches for a given query. In this part we will learn, how to search exactly, allow substitutions and indels, and how to configure what kind of results we want, e.g. all results vs. only the best result.
+
+## Terminology
+### Search scheme
+A search scheme is the manner of how a search is conducted. Depending on what kind of errors are allowed, there is an optimal way of search. For example, for 1 allowed substitution the trivial approach is to search for all possible permutations with one substitution. It is more efficient to divide the query into two blocks, search one block exactly and allow the error in the other block. Even though this has to be done twice (error in block 1 and error in block 2), this requires less lookups in the index than the trivial approach.
+### Optimal Search Scheme
+The scheme that requires the least amount of queries in the index is called an *optimal search scheme* and determining it is NP-hard.<br>
+SeqAn3 provides *optimal* search schemes for up to **4** errors for Bi-FM-Indices and provides sub-optimal, but still efficient search schemes otherwise.<br>
+The choice of the search scheme is done internally, but when using >= 2 errors it is advisable to construct a Bi-FM-Index since searches will be faster.<br>
+You can read more about search schemes in [this paper](http://doi.org/10.1101/301085).
+
+## Example
+
+We can search for all exact matches using seqan3::search:
+
+\include doc/tutorial/search/basic_search.cpp
+
+When using a single text, the search will return a std::vector of the begin positions of matches in the reference.
+For text collections, a tuple where the first element represents the index of the text in the collection and the second element represents the position within the text.<br>
+You can also pass multiple queries at the same time:
+
+\snippet doc/tutorial/search/small_snippets.cpp multiple_queries
+
+The returned result will be a vector of individual results. Each element refers to the respective query.
+
+\assignment{Assignment 2}
+Search for all exact occurrences of `GCT` in the text from assignment 1.<br>
+Print the number of hits and the positions in a ascending ordering.<br>
+Do the same for the following text collection:
+\code
+std::vector<dna4_vector> text{"CGCTGTCTGAAGGATGAGTGTCAGCCAGTGTA"_dna4,
+                              "ACCCGATGAGCTACCCAGTAGTCGAACTG"_dna4,
+                              "GGCCAGACAACCCGGCGCTAATGCACTCA"_dna4};
+\endcode
+\endassignment
+
+\solution
+\include doc/tutorial/search/solution2.cpp
+**Expected output:**
+~~~console
+=====   Running on a single text   =====
+There are 3 hits.
+The positions are [1,41,77]
+===== Running on a text collection =====
+There are 3 hits.
+The positions are [(0,1),(1,9),(2,16)]
+~~~
+\endsolution
+
+## Configuration
+
+We offer the following configurations for allowing errors:
```

Comment:
Something along the lines
```suggestion
If you want to allow for errors in your query, you need to configure the approximate search with a seqan3::search_cfg::max_error or seqan3::search_cfg::max_error_rate object. The is constructed with four parameters:
```
Question: are all  combinations of numbers possible?